controller_manager/test/test_controller_manager.cpp

```diff
@@ -124,3 +126,80 @@ TEST_F(TestControllerManager, controller_lifecycle) {
     test_controller->get_lifecycle_node()->get_current_state().id());
   EXPECT_EQ(1, test_controller.use_count());
 }
+
+TEST_F(TestControllerManager, controller_parameters) {
+  auto cm = std::make_shared<controller_manager::ControllerManager>(
+    robot_, executor_,
+    "test_controller_manager");
+
+  auto executor_spin_future = std::async(
+    std::launch::async, [this]() -> void {
+      executor_->spin();
+    });
+  // This sleep is needed to prevent a too fast test from ending before the
+  // executor has began to spin, which causes it to hang
+  std::this_thread::sleep_for(50ms);
```

Comment:
I know, I haven't found any simple one. I was generous with the sleep (I needed a fraction of those 50ms), but I don't like either.

To me it's a bug, because the destructor of the spin future hangs without this every now and then.

There's nothing that I see on the Executor API to check if it's spinning. A way would be to add a timer or something and wait until it is triggered by the executor, but it's not going to be pretty.

Reply:
FYI I reported https://github.com/ros2/rclcpp/issues/1454
And attempted the workaround I suggested, which lead to a segfault, reported in https://github.com/ros2/rclcpp/issues/1455

For now I'm gonna leave the sleep here and we can revisit this in the future if we manage to fix this upstream.

Reply:
Do you think we can accept this as it is, as the rclcpp might be a long way ahead? 
I don't like the sleeps either, specially since I was pushing changes to address https://github.com/ros-controls/ros2_controllers/issues/27 but I can't find a nice solution to this right now.